src/main/java/uk/gov/pay/ledger/transaction/resource/TransactionResource.java

```diff
@@ -100,6 +108,47 @@ public TransactionSearchResponse search(@Valid
         return transactionService.searchTransactionsForCsv(searchParams);
     }
 
+    @Path("/stream")
+    @GET
+    @Produces("text/csv")
+    @Timed
+    public Response streamCsv(@Valid @BeanParam TransactionSearchParams searchParams,
+                              @QueryParam("override_account_id_restriction") Boolean overrideAccountRestriction,
+                              @QueryParam("account_id") String gatewayAccountId,
+                              @Context UriInfo uriInfo) {
+        StreamingOutput stream = outputStream -> {
+            TransactionSearchParams csvSearchParams = Optional.ofNullable(searchParams).orElse(new TransactionSearchParams());
+            csvSearchParams.overrideMaxDisplaySize(1000L);
+            csvSearchParams.setAccountId(gatewayAccountId);
+            List<TransactionEntity> page;
+            ZonedDateTime startingAfterCreatedDate = null;
+            Long startingAfterId = null;
+            int count = 0;
+
+            Map<String, Object> headers = csvService.csvHeaderFrom(null);
+            ObjectWriter writer = csvService.writerFrom(headers);
+
+            outputStream.write(csvService.csvStringFrom(headers, writer).getBytes());
+            do {
+                page = transactionService.searchTransactionAfter(csvSearchParams, startingAfterCreatedDate, startingAfterId);
+                count += page.size();
+
+                if (!page.isEmpty()) {
+                    var lastEntity = page.get(page.size() - 1);
+                    startingAfterCreatedDate = lastEntity.getCreatedDate();
+                    startingAfterId = lastEntity.getId();
+
+                    outputStream.write(
+                            csvService.csvStringFrom(page, writer).getBytes()
+                    );
+                    outputStream.flush();
+                }
+            } while (!page.isEmpty() && count < 200000);
```

Comment:
Just a quick note that `count` should be multiple of value set for `overrideMaxDisplaySize`. If not there is a chance we deliver more records than the limit